webapp/cas-server-webapp-resources/src/main/resources/log4j2.xml

```diff
@@ -56,69 +88,76 @@
+        <Logger name="org.springframework.web.socket" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.springframework.boot.diagnostics.LoggingFailureAnalysisReporter" level="trace" includeLocation="${sys:log.include.location}"/>
 
-        <AsyncLogger name="org.opensaml" level="${sys:opensaml.log.level}" includeLocation="true"/>
-        <AsyncLogger name="PROTOCOL_MESSAGE" level="${sys:opensaml.log.level}" includeLocation="true" />
+        <Logger name="com.couchbase" level="warn" includeLocation="${sys:log.include.location}" />
+        <Logger name="org.apache" level="error" includeLocation="${sys:log.include.location}"/>
+        <Logger name="com.netflix" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.quartz" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.thymeleaf" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.pac4j" level="${sys:pac4j.log.level}" includeLocation="${sys:log.include.location}"/>
 
-        <AsyncLogger name="net.sf.ehcache" level="warn" includeLocation="true"/>
-        <AsyncLogger name="net.jradius" level="warn" includeLocation="true"/>
-        <AsyncLogger name="org.openid4java" level="warn" includeLocation="true"/>
-        <AsyncLogger name="org.ldaptive" level="${sys:ldap.log.level}" includeLocation="true"/>
-        <AsyncLogger name="com.hazelcast" level="${sys:hazelcast.log.level}" includeLocation="true"/>
-        <AsyncLogger name="org.jasig.spring" level="warn" includeLocation="true"/>
-        <AsyncLogger name="org.apache.cxf" level="warn" includeLocation="true"/>
-        <AsyncLogger name="org.apache.http" level="warn" includeLocation="true"/>
+        <Logger name="org.opensaml" level="${sys:opensaml.log.level}" includeLocation="${sys:log.include.location}"/>
+        <Logger name="PROTOCOL_MESSAGE" level="${sys:opensaml.log.level}" includeLocation="${sys:log.include.location}" />
 
-        <AsyncRoot level="warn">
+        <Logger name="net.sf.ehcache" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="net.jradius" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.openid4java" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.ldaptive" level="${sys:ldap.log.level}" includeLocation="${sys:log.include.location}"/>
+        <Logger name="com.hazelcast" level="${sys:hazelcast.log.level}" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.jasig.spring" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.apache.cxf" level="warn" includeLocation="${sys:log.include.location}"/>
+        <Logger name="org.apache.http" level="warn" includeLocation="${sys:log.include.location}"/>
+b
+        <Root level="warn">
```

Comment:
Hi, I wonder if you also managed to make location info like line numbers (`%L` in a pattern) printed when _async_ is activated? Because according to the [docs](https://logging.apache.org/log4j/2.x/manual/async.html), you need to:

> set "includeLocation=true" in the configuration of all relevant loggers, including the root logger"

And even the Log4j documentation doesn't seem to be quite correct, because according to my experiments, the `includeLocation` is inherited from the root to children loggers and trying to override the value doesn't have any effect. So in short, specifying `includeLocation` at individual loggers doesn't make much sense under these circumstances.